lib/types/src/transaction/encode.rs

```diff
@@ -126,7 +120,7 @@ impl From<L2Envelope> for TransactionData {
             from: Address::ZERO,
             to: l2_tx.to().unwrap_or_default(),
             gas_limit: U256::from(l2_tx.gas_limit()),
-            pubdata_price_limit: U256::from(gas_per_pubdata_limit),
+            pubdata_price_limit: U256::from(0),
```

Comment:
This change is strange. The TransactionData already had a field called "pubdata_price_limit"? How come?

Reply:
We used to pass 50k here but one of the recently added changes asserts that `gas_per_pubdata` is zero for all L2 transactions. The field name itself is likely inaccurate if that's what you mean, but we need to migrate to using zksync_os_api for encoding regardless.